examples/rag_bm.py

```diff
@@ -0,0 +1,271 @@
+                        ranker_configs=[LLMRankerConfig()],
+                        retriever_configs=[FAISSRetrieverConfig(persist_path=output_dir), BM25RetrieverConfig()],
+                    )
+                else:
+                    logger.info("Loading index from document!")
+                    self.engine = SimpleEngine.from_docs(
+                        input_files=dataset.document_files,
+                        retriever_configs=[FAISSRetrieverConfig(persist_path=output_dir), BM25RetrieverConfig()],
+                        ranker_configs=[FlagEmbeddingConfig()],
+                        transformations=[SentenceSplitter(chunk_size=256, chunk_overlap=0)],
+                    )
+                results = []
+                for gt_info in dataset.gt_info:
+                    result = await self.rag_evaluate_single(
+                        question=gt_info["question"],
+                        reference=gt_info["gt_reference"],
+                        ground_truth=gt_info["gt_answer"],
+                    )
+                    results.append(result)
+                logger.info(f"=====The {dataset.name} BenchMark dataset assessment is complete!=====")
+                self._print_bm_result(results)
+
+                write_json_file(os.path.join(EXAMPLE_BENCHMARK_PATH, dataset.name, "bm_result.json"), results, "utf-8")
+
+    async def rag_evaluate_single(self, question, reference, ground_truth, print_title=True):
+        """This example run rag pipeline, use faiss&bm25 retriever and llm ranker, will print something like:
+
+        Retrieve Result:
+        0. Productivi..., 10.0
+        1. I wrote cu..., 7.0
+        2. I highly r..., 5.0
+
+        Query Result:
+        Passion, adaptability, open-mindedness, creativity, discipline, and empathy are key qualities to be a good writer.
+
+        RAG BenchMark result:
+        {
+        'metrics':
+         {
+         'bleu-avg': 0.48318624982047,
+         'bleu-1': 0.5609756097560976,
+         'bleu-2': 0.5,
+         'bleu-3': 0.46153846153846156,
+         'bleu-4': 0.42105263157894735,
+         'rouge-L': 0.6865671641791045,
+         'semantic similarity': 0.9487444961487591,
+         'length': 74
+         },
+         'log': {
+         'generated_text':
+         '国家卫生健康委在2023年7月28日开展的“启明行动”是为了防控儿童青少年的近视问题。活动发布的指导性文件名称为《防控儿童青少年近视核心知识十条》。',
+         'ground_truth_text':
+         '“启明行动”是为了防控儿童青少年的近视问题，并发布了《防控儿童青少年近视核心知识十条》。'
+            }
+         }
+        """
+        if print_title:
+            self._print_title("RAG Pipeline")
+        try:
+            nodes = await self.engine.aretrieve(question)
+            #    pdb.set_trace()
+            self._print_result(nodes, state="Retrieve")
+
+            answer = await self.engine.aquery(question)
+            self._print_result(answer, state="Query")
+
+        except Exception as e:
+            print(e)
+            return {
+                "metrics": {

```

Comment:
repeat with Line 196-214, can you make it a single method like `RAGBenchmark._set_metrics` with default value, and return desired result if pass different values.
and why not put it (Line 190-194 and so on) into `RAGBenchmark.compute_metric` so you can get the all metrics with one method calling.